include/fiction/io/write_sqd_layout.hpp

```diff
@@ -273,15 +283,31 @@ class write_sqd_layout_impl
             lyt.foreach_sidb_defect(
                 [&design](const auto& cd)
                 {
-                    const auto& cell   = cd.first;
                     const auto& defect = cd.second;
 
-                    design << fmt::format(siqad::DEFECT_BLOCK,
-                                          fmt::format(siqad::LATTICE_COORDINATE, cell.x, cell.y / 2, cell.y % 2),
-                                          is_charged_defect(defect) ? fmt::format(siqad::COULOMB, defect.charge,
-                                                                                  defect.epsilon_r, defect.lambda_tf) :
-                                                                      "",
-                                          get_defect_type_name(defect.type));
+                    // layout is not based on siqad coordinates, coordinate transformation is performed
+                    if constexpr (!has_siqad_coord_v<Lyt>)
```

Comment:
Please switch the branches. Start with the case that the layout has SiQAD coordinates. This avoids the double negation necessary to comprehend the else branch as a human ("if the layout does not not have SiQAD coords...")

Reply:
Absolutely! Sorry, you told me that a few months ago. And I am aware of it. However, it was required in a previous structure that didn't work. Then I forgot to change it back. Sorry, I will fix it.